src/vm-repair/azext_vm_repair/repair_utils.py

```diff
@@ -407,6 +429,116 @@ def _fetch_compatible_windows_os_urn(source_vm):
+    elif distro == 'centos8':
+        os_image_urn = 'OpenLogic:CentOS:8_4:latest'
+    elif distro == 'oracle6':
+        os_image_urn = 'Oracle:Oracle-Linux:6.10:latest'
+    elif distro == 'oracle7':
+        os_image_urn = 'Oracle:Oracle-Linux:ol79:latest'
+    elif distro == 'oracle8':
+        os_image_urn = 'Oracle:Oracle-Linux:ol82:latest'
+    elif distro == 'sles12':
+        os_image_urn = _suse_image_selector('sles-12')
+    elif distro == 'sles15':
+        os_image_urn = _suse_image_selector('sles-15')
+    else:
+        if distro.count(":") == 3:
+            logger.info('A custom URN was provided , will be used as distro for the recovery VM')
+            os_image_urn = distro
+        else:
+            logger.info('No specific distro was provided , using the default Ubuntu distro')
+            os_image_urn = "UbuntuLTS"
+    return os_image_urn
+
+
+def _select_distro_linux_gen2(distro):
+    # base on the document : https://docs.microsoft.com/en-us/azure/virtual-machines/generation-2#generation-2-vm-images-in-azure-marketplace
+    # RHEL/Centos/Oracle 6 are not supported for Gen 2
+    if distro == 'rhel6':
+        os_image_urn = 'RedHat:rhel-raw:7-raw-gen2:latest'
+    elif distro == 'rhel7':
+        os_image_urn = 'RedHat:rhel-raw:7-raw-gen2:latest'
+    elif distro == 'rhel8':
+        os_image_urn = 'RedHat:rhel-raw:8-raw-gen2:latest'
+    elif distro == 'ubuntu18':
+        os_image_urn = 'Canonical:UbuntuServer:18_04-lts-gen2:latest'
+    elif distro == 'ubuntu20':
+        os_image_urn = "Canonical:0001-com-ubuntu-server-focal:20_04-lts-gen2:latest"
+    elif distro == 'centos6':
+        os_image_urn = 'OpenLogic:CentOS:7_9-gen2:latest'
+    elif distro == 'centos7':
+        os_image_urn = 'OpenLogic:CentOS:7_9-gen2:latest'
+    elif distro == 'centos8':
+        os_image_urn = 'OpenLogic:CentOS:8_4-gen2:latest'
+    elif distro == 'oracle6':
+        os_image_urn = 'Oracle:Oracle-Linux:ol79-gen2:latest'
+    elif distro == 'oracle7':
+        os_image_urn = 'Oracle:Oracle-Linux:ol79-gen2:latest'
+    elif distro == 'oracle8':
+        os_image_urn = 'Oracle:Oracle-Linux:ol82-gen2:latest'
+    elif distro == 'sles12':
+        os_image_urn = _suse_image_selector_gen2('sles-12')
+    elif distro == 'sles15':
+        os_image_urn = _suse_image_selector_gen2('sles-15')
+    else:
+        if distro.count(":") == 3:
+            logger.info('A custom URN was provided , will be used as distro for the recovery VM')
+            if distro.find('gen2'):
+                os_image_urn = distro
+            else:
+                logger.info('The provided URN does not contain Gen2 in it and this VM is a gen2 , dropping to default image')
+                os_image_urn = "Canonical:UbuntuServer:18_04-lts-gen2:latest"
+        else:
+            logger.info('No specific distro was provided , using the default Ubuntu distro')
+            os_image_urn = "Canonical:UbuntuServer:18_04-lts-gen2:latest"
+    return os_image_urn
```

Comment:
Same as above

Reply:
Thank you for the update and notes :) 
sure, I will go over the comments and do the  needed (Y)